### DISCUSSION TITLE
Can I use quackosm to accelerate the feature extract step in the  "Transfer learning approach to bicycle sharing systems station location planning using OpenStreetMap" project?

### DISCUSSION THREAD

**Discussion by zrg1993** [kraina-ai/quackosm#161] [General]
Thanks for your work "**_Transfer learning approach to bicycle sharing systems station location planning using OpenStreetMap_**". 
And I wonder that if quackosm can speed up the feature extraction in this project.

**Comment by RaczeQ** [comment]
Hello @zrg1993 👋🏻 
You absolutely can!
If you could use our other library - [SRAI](https://github.com/kraina-ai/srai), then you would have a lot of tools available for you to prepare the data for the training. SRAI uses QuackOSM under the hood, so you won't have to worry about the speed.

What you would need is a `ContextualCountEmbedder` (or plain `CountEmbedder`) that was used in this publication. Here is a full usage documentation with OSM data download: https://kraina-ai.github.io/srai/0.7.5/examples/embedders/contextual_count_embedder/

Here you can see a full example of this transfer learning pipeline in some notebooks from our tutorials:
https://github.com/kraina-ai/srai-tutorial/blob/ml-in-pl-2023/tutorial/MLinPL/05_bikes_transfer_learning.ipynb - this one you can use as a training notebook with answers in another directory.
https://github.com/kraina-ai/srai-tutorial/blob/osm-deep-dive/tutorial/06_use_osm_data_in_ml_model.ipynb
https://github.com/kraina-ai/srai-tutorial/blob/sotm2024/03_ml.ipynb

**Comment by zrg1993** [comment]
@RaczeQ Thank you so much. I am checking the tutorial. If I success it, I will notify you.
